tortoise/README.md

```diff
@@ -0,0 +1,167 @@
+Tortoise is used to make decisions on the blocks that will be applied to the ledger. The protocol is asynchronous and makes progress based on the received data, without timing assumptions.
+
+## Core concepts
+
+#### Block
+
+From the tortoise perspective it is important that the block has a unique ID and belongs to some layer.
+
+In the rest of the system block is a wrapper for transactions and rewards. So once tortoise decides on a block - related transactions and rewards will be applied to the ledger.
+
+#### Ballot
+
+Tortoise counts votes from the syntactically valid ballots.
+
+For tortoise purposes every ballot must have a unique ID, reference to the base ballot ID, explicit votes (exceptions), weight and a beacon.
+
+###### Base ballot and exceptions
+
+Discussed in [Voting section](#voting).
+
+###### Weight
+
+Weight of the ballot is computed from the weight of the activation transaction divided by the total number of eligible ballots. Total number of eligible ballots is computed based on the fraction of actiovation transaction compared to the weight from the active set that was recorded in reference ballot.
+
+###### Beacon
+
+Ballot with an inconsistent beacon will not be marked good and will not be counted immediatly. Discussed in [full](#votes-from-ballots-with-wrong-beacon) and [verifying](#good-ballots) sections.
+
+#### Expected weight and thresholds
+
+Expected weight is a total weight from the activation transactions. Unlike weight for the ballot we are using a sum of all activation transactions that target a specific epoch. Expected weight for the layer is fraction of the expected weight for the epoch.
+
+## Voting
+
+When voting protocol chooses base ballot and encodes the difference between local opinion and base ballot votes. Since we don't keep votes from genesis in memory the implementation has a shortcoming that it encodes difference only within some configurable window (we can refer to it as voting window). 
+
+In order to minimize amount of data that needs to be sent and stored, Tortoise selects ballot that has the least difference with a local opinion. 
+
+Types of votes:
+- support
+
+    Support all blocks according to local opinion that are not supported by base ballot. Usually support is added for layers starting from base ballot layer up to last layer. In the event of healing tortoise may add support for blocks before base ballot layer.
+
+- against
+  
+    Against is added only if the base ballot supports the block and local opinion is against that block. 
```

Comment:
Why/under what circumstances might this happen?

Reply:
this depends on the strategy for choosing base ballot. it might be more efficient to choose a ballot from later layer, and vote against blocks from previous layers. we are not doing this kind of optimizations